external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutConfig.java

```diff
@@ -230,8 +237,37 @@ public Builder(Map<String, Object> kafkaProps, KafkaSpoutStreams kafkaSpoutStrea
         }
 
         public KafkaSpoutConfig<K,V> build() {
+            validate();
             return new KafkaSpoutConfig<>(this);
         }
+
+        /**
+         * Defines whether the consumer manages partition manually.
+         * If set to true, the consumer behaves like a simple consumer, otherwise it will rely on kafka to do partition assignment.
+         * @param manualPartitionAssign Whether use manual partition assignment.
+         */
+        public Builder setManualPartitionAssign(boolean manualPartitionAssign) {
+            this.manualPartitionAssign = manualPartitionAssign;
+            return this;
+        }
+
+        /**
+         * Defines partition refresh period in the manual partition assign model.
+         * @param partitionRefreshPeriodMs Partition refresh period in ms.
+         */
+        public Builder setPartitionRefreshPeriodMs(int partitionRefreshPeriodMs) {
+            this.partitionRefreshPeriodMs = partitionRefreshPeriodMs;
+            return this;
+        }
+
+        /**
+         * Validate configs before build.
+         */
+        private void validate() {
+            if (this.manualPartitionAssign && kafkaSpoutStreams instanceof KafkaSpoutStreamsWildcardTopics) {
```

Comment:
Probably better to check that kafkaSpoutStreams is an instance of KafkaSpoutStreamsNamedTopics, so this doesn't break if more subclasses are added at some point.